### PR TITLE
SPLAT-1620: migrate to vsphere-elastic pool

### PR DESCRIPTION
**Pull request by rvanderp3**
*What is the vsphere-elastic pool?*  The vsphere-elastic pool represents an aggregate of all vSphere CI capacity.  Rather than this capacity being segmented in to different cluster profiles, and thus possibly overloading some environments while others are idle.  The vSphere elastic-pool relies upon a [scheduler](https://github.com/openshift-splat-team/vsphere-capacity-manager) that determines the most appropriate environment for a given CI job.

Not only does this allow hardware to be more fully and appropriately utilized, it allows more streamlined maintenance of resources.  Jobs, by default, will be allocated to the least busy environment as determined by leases.  If a job must land on a specific environment, that can be set with the environment variable `POOLS`.  To see which pools are available, one can run `oc get pools.vspherecapacitymanager.splat.io -n vsphere-infra-helpers` on build02.

### CODE REVIEW

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn

File: ci-operator/step-registry/ipi/conf/vsphere/check/vcm/ipi-conf-vsphere-check-vcm-commands.sh [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    job-name: "${JOB_NAME_SAFE}"
+spec:
+  vcpus: ${OPENSHIFT_REQUIRED_CORES}
+  memory: ${OPENSHIFT_REQUIRED_MEMORY}
+  ${requiredPool}
+  networks: 1" | oc create --kubeconfig ${SA_KUBECONFIG} -f -
+done
+
+log "waiting for lease to be fulfilled..."
+oc wait leases.vspherecapacitymanager.splat.io --kubeconfig ${SA_KUBECONFIG} --timeout=30m --for=jsonpath='{.status.phase}'=Fulfilled -n vsphere-infra-helpers -l boskos-lease-id="${LEASED_RESOURCE}"
```
Comment (jcpowermac) [inline ci-operator/step-registry/ipi/conf/vsphere/check/vcm/ipi-conf-vsphere-check-vcm-commands.sh:237]: default is 30 seconds, what happens if there are no resources available in a pool?

Comment (jcpowermac) [reply ci-operator/step-registry/ipi/conf/vsphere/check/vcm/ipi-conf-vsphere-check-vcm-commands.sh:237]: nm lol

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-installer-master-e2e-vsphere-zones pull-ci-openshift-installer-release-4.16-e2e-vsphere-zones pull-ci-openshift-installer-release-4.17-e2e-vsphere-zones  pull-ci-openshift-installer-release-4.16-e2e-vsphere-ovn  pull-ci-openshift-installer-release-4.17-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-ovn periodic-ci-openshift-release-master-nightly-4.17-e2e-vsphere-zones

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator pull-ci-openshift-installer-master-e2e-vsphere-zones

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator pull-ci-openshift-installer-master-e2e-vsphere-zones

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-upgrade-from-stable-4.14-e2e-vsphere-upgrade

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.15-upgrade-from-stable-4.14-e2e-vsphere-upgrade

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator pull-ci-openshift-installer-master-e2e-vsphere-zones pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere-csi pull-ci-openshift-okd-machine-os-master-e2e-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator pull-ci-openshift-installer-master-e2e-vsphere-zones pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere-csi pull-ci-openshift-okd-machine-os-master-e2e-vsphere

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-priv-cluster-api-provider-vsphere-release-4.16-regression-clusterinfra-vsphere-ipi-techpreview pull-ci-openshift-cluster-api-provider-vsphere-release-4.17-regression-clusterinfra-vsphere-ipi-techpreview pull-ci-openshift-installer-release-4.16-e2e-vsphere-zones periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-zones pull-ci-openshift-priv-cloud-provider-vsphere-release-4.16-e2e-vsphere-ovn pull-ci-openshift-priv-installer-release-4.16-e2e-vsphere-ovn pull-ci-openshift-assisted-installer-master-edge-e2e-vsphere-assisted periodic-ci-openshift-release-master-ci-4.16-e2e-vsphere-ovn-upgrade pull-ci-openshift-machine-api-operator-release-4.16-e2e-vsphere-operator

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-vsphere-assisted

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-vsphere-assisted

**Comment by rvanderp3** [comment]
pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator pull-ci-openshift-installer-master-e2e-vsphere-zones pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere-csi pull-ci-openshift-okd-machine-os-master-e2e-vsphere pull-ci-openshift-installer-release-4.16-e2e-vsphere-upi-zones pull-ci-openshift-installer-release-4.16-e2e-vsphere-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-zones

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-vsphere-assisted pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator pull-ci-openshift-installer-master-e2e-vsphere-zones pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere-csi pull-ci-openshift-okd-machine-os-master-e2e-vsphere pull-ci-openshift-installer-release-4.16-e2e-vsphere-upi-zones pull-ci-openshift-installer-release-4.16-e2e-vsphere-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-zones

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-assisted-installer-master-edge-e2e-vsphere-assisted pull-ci-openshift-cloud-provider-vsphere-master-e2e-vsphere-ovn pull-ci-openshift-machine-api-operator-release-4.17-e2e-vsphere-operator pull-ci-openshift-machine-config-operator-master-e2e-vsphere pull-ci-openshift-vsphere-problem-detector-master-e2e-vsphere pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere pull-ci-openshift-installer-master-e2e-vsphere-ovn pull-ci-openshift-cluster-control-plane-machine-set-operator-main-e2e-vsphere-operator pull-ci-openshift-installer-master-e2e-vsphere-zones pull-ci-openshift-vmware-vsphere-csi-driver-master-e2e-vsphere-csi pull-ci-openshift-okd-machine-os-master-e2e-vsphere pull-ci-openshift-installer-release-4.16-e2e-vsphere-upi-zones pull-ci-openshift-installer-release-4.16-e2e-vsphere-upi periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-zones

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-zones periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn pull-ci-openshift-machine-config-operator-release-4.16-e2e-vsphere pull-ci-openshift-machine-config-operator-release-4.16-e2e-vsphere-ovn-zones periodic-ci-openshift-release-master-nightly-4.17-upgrade-from-stable-4.16-e2e-vsphere-upgrade periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-static-ovn periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-techpreview-serial

**Comment by jcpowermac** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-zones periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn pull-ci-openshift-machine-config-operator-release-4.16-e2e-vsphere pull-ci-openshift-machine-config-operator-release-4.16-e2e-vsphere-ovn-zones periodic-ci-openshift-release-master-nightly-4.17-upgrade-from-stable-4.16-e2e-vsphere-upgrade periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-static-ovn periodic-ci-openshift-release-master-nightly-4.16-e2e-vsphere-ovn-techpreview-serial

**Review by r4f4 — COMMENTED** [review]
/approve

**Comment by r4f4** [comment]
Let's make sure older releases are not impacted:

/pj-rehearse pull-ci-openshift-installer-release-4.12-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.13-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.14-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
> Let's make sure older releases are not impacted:
> 
> /pj-rehearse pull-ci-openshift-installer-release-4.12-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.13-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.14-e2e-vsphere-ovn

good call @r4f4 , we have a bug there.  im placing a hold until it's fixed.

**Comment by sinnykumari** [comment]
For MCO related changes
/approve

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.12-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.13-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.14-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-installer-release-4.12-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.13-e2e-vsphere-ovn pull-ci-openshift-installer-release-4.14-e2e-vsphere-ovn

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-serial

**Comment by rvanderp3** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-e2e-vsphere-ovn-serial

**Comment by rvanderp3** [comment]
/pj-rehearse pull-ci-openshift-windows-machine-config-operator-release-4.17-vsphere-e2e-operator pull-ci-openshift-windows-machine-config-operator-release-4.17-platform-none-vsphere-e2e-operator

**Comment by rvanderp3** [comment]
/hold

placing hold for over the weekend so we can watch things after merge on monday

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.14-installer-rehearse-vsphere

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-vsphere-ipi-proxy-fips-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-vsphere-upi-encrypt-f14

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-vsphere-upi-platform-none-f28

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-vsphere-ipi-customized-resource-f28

**Pull request merged**